test/integration/scheduler/queue_test.go

```diff
@@ -713,14 +713,14 @@ func TestCoreResourceEnqueue(t *testing.T) {
 			},
 			triggerFn: func(testCtx *testutils.TestContext) error {
 				// Trigger an assigned Node update event.
-				// It should requeue pod4 only because this node only has zone label, and doesn't have node label that pod3 requires.
+				// It should requeue pod3, pod4, the update adds the "zone" label to fake-node2.
 				node := st.MakeNode().Name("fake-node2").Label("zone", "fake-node").Obj()
 				if _, err := testCtx.ClientSet.CoreV1().Nodes().Update(testCtx.Ctx, node, metav1.UpdateOptions{}); err != nil {
 					return fmt.Errorf("failed to update node: %w", err)
 				}
 				return nil
 			},
-			wantRequeuedPods:          sets.New("pod4"),
+			wantRequeuedPods:          sets.New("pod3", "pod4"),
```

Comment:
https://github.com/kubernetes/kubernetes/pull/127478#discussion_r1770759393
This seems to be the scenario of not match -> match and match -> not match

Reply:
Why does this test case requeue both pod3 and 4?
The addition of zone isn't related to pod3, right? because pod3 has topologyKey:node, not zone.

Reply:
Ah, okay got it. I missed that the update removes the node label from the node